### ISSUE TITLE
cluster-stamp.bicep produces some warnings on deployment

### ISSUE THREAD

**Issue by akulich** [mspnp/aks-baseline#314]
Here are the warnings produced by `cluster-stamp.bicep` deployment (line numbers are slightly different as it was modified with couple of additional parameters):
```
cluster-stamp.bicep(178,18) : Warning BCP081: Resource type "Microsoft.ContainerService/managedClusters/namespaces@2022-01-02-preview" does not have types available.
cluster-stamp.bicep(217,5) : Warning BCP037: The property "etag" is not allowed on objects of type "SavedSearchProperties". Permissible properties include "functionAlias", "functionParameters", "tags". If this is an inaccuracy in the documentation, please report it to the Bicep Team. [https://aka.ms/bicep-type-issues]
cluster-stamp.bicep(230,5) : Warning BCP037: The property "etag" is not allowed on objects of type "SavedSearchProperties". Permissible properties include "functionAlias", "functionParameters", "tags". If this is an inaccuracy in the documentation, please report it to the Bicep Team. [https://aka.ms/bicep-type-issues]
cluster-stamp.bicep(907,5) : Warning BCP073: The property "scope" is read-only. Expressions cannot be assigned to read-only properties. If this is an inaccuracy in the documentation, please report it to the Bicep Team. [https://aka.ms/bicep-type-issues]
cluster-stamp.bicep(932,5) : Warning BCP073: The property "scope" is read-only. Expressions cannot be assigned to read-only properties. If this is an inaccuracy in the documentation, please report it to the Bicep Team. [https://aka.ms/bicep-type-issues]
cluster-stamp.bicep(951,5) : Warning BCP073: The property "scope" is read-only. Expressions cannot be assigned to read-only properties. If this is an inaccuracy in the documentation, please report it to the Bicep Team. [https://aka.ms/bicep-type-issues]
cluster-stamp.bicep(970,5) : Warning BCP073: The property "scope" is read-only. Expressions cannot be assigned to read-only properties. If this is an inaccuracy in the documentation, please report it to the Bicep Team. [https://aka.ms/bicep-type-issues]
cluster-stamp.bicep(994,5) : Warning BCP073: The property "scope" is read-only. Expressions cannot be assigned to read-only properties. If this is an inaccuracy in the documentation, please report it to the Bicep Team. [https://aka.ms/bicep-type-issues]
cluster-stamp.bicep(1024,5) : Warning BCP073: The property "scope" is read-only. Expressions cannot be assigned to read-only properties. If this is an inaccuracy in the documentation, please report it to the Bicep Team. [https://aka.ms/bicep-type-issues]
cluster-stamp.bicep(1051,5) : Warning BCP073: The property "scope" is read-only. Expressions cannot be assigned to read-only properties. If this is an inaccuracy in the documentation, please report it to the Bicep Team. [https://aka.ms/bicep-type-issues]
cluster-stamp.bicep(1386,5) : Warning BCP073: The property "maxAgentPools" is read-only. Expressions cannot be assigned to read-only properties. If this is an inaccuracy in the documentation, please report it to the Bicep Team. [https://aka.ms/bicep-type-issues]
```

It doesn't prevent cluster from deploying. Mostly FYI.

Running v1.23.5.0 / latest main branch

**Comment by ckittel** [comment]
You are indeed correct that none of these warnings are blockers to the deployment.  Thanks for bringing it up.

- `etag` -- that's tracked in Bicep @ https://github.com/Azure/bicep/issues/5256.  Safe to ignore.
- `"Microsoft.ContainerService/managedClusters/namespaces@2022-01-02-preview" does not have types available` - That's expected, there are no types available for this Azure resource. Safe to ignore and will go away once the `namespace` child resource gets types added.
- the `scope` one has been reported upstream at https://github.com/Azure/bicep/issues/784#issuecomment-721376521, but I honestly think we can avoid the warning with a change to the bicep.  But in the mean time, safe to ignore.
- the `maxAgentPools` is worth looking into.  I'm pretty sure that is indeed a read only property (I don't think it always was though).  Either way, it's safe to ignore because if it is read only, then the resource provider will ignore it. If it's not read only, it's ideally the value we'd want it set at :)

We'll keep this issue open to track both `scope` and `maxAgentPools` -- but we cannot influence the other two, without adding warning ignore statements on those lines -- which we could consider adding.

**Comment by ckittel** [comment]
There are no more warnings on the deployment.  They have all been resolved as part of the work in #317.  Thanks again for caring enough to encourage us to take action.  <3
